inc/cookie/namespace.php

```diff
@@ -29,10 +29,25 @@ function on_login_form() {
  * Load hook to check for the oauth2 redirect request.
  */
 function on_load() {
-	if ( wp_parse_url( $_SERVER['REQUEST_URI'], PHP_URL_PATH ) !== '/hm-delegated-auth-callback' ) {
+	$home_path = wp_parse_url( home_url(), PHP_URL_PATH );
+	$req_path = wp_parse_url( $_SERVER['REQUEST_URI'], PHP_URL_PATH );
+	if ( substr( $req_path, strlen( $home_path ) ) !== '/hm-delegated-auth-callback' ) {
```

Comment:
Took a sec to grok this! I think we could end up with a failed match if `home_url()` returned a slash. It's filtered so not sure we can trust it

Reply:
I tested with subdomain and subdirectory installs including at the root, so don't think so?